File: 0958-check-completeness-of-a-binary-tree/0958-check-completeness-of-a-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public bool IsCompleteTree(TreeNode root) {
           Queue<TreeNode> q = new Queue<TreeNode>();
    q.Enqueue(root);

    bool isNull = false;

    while (q.Count > 0)
    {
        TreeNode temp = q.Dequeue();

        if (temp == null)
        {
            isNull = true;
        }
        else
        {
            if (isNull)
            {
                return false;
            }

            q.Enqueue(temp.left);
            q.Enqueue(temp.right);
        }
    }

    return true;
    }
}

File: 0958-check-completeness-of-a-binary-tree/0958-check-completeness-of-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isCompleteTree = function(root) {
     let queue = [];
    queue.push(root);

    let isNULL = false;

    while (queue.length > 0) {
        let temp = queue.shift();

        if (temp === null) {
            isNULL = true;
        } else {
            if (isNULL === true) {
                return false;
            }

            queue.push(temp.left);
            queue.push(temp.right);
        }
    }

    return true;
};